app/assets/javascripts/directives/search_sidebar.js

```diff
@@ -0,0 +1,170 @@
+
+    controller: function($scope) {
+      $scope.searching = false; // currently performing a seach
+      $scope.term = ""; // what the user has searched for
+      $scope.count = 0; // number of results returned
+      $scope.morePages = false; // the search results include more pages
+      $scope.page = 0; // current page of results
+      $scope.loadingMore = false; // load more function is running
+
+      // Result templates
+      $scope.project_template = JST['templates/search/project'];
+      $scope.story_template = JST['templates/search/story'];
+      $scope.task_template = JST['templates/search/task'];
+      $scope.comment_template = JST['templates/search/comment'];
+      $scope.attachment_template = JST['templates/search/attachment'];
+
+      /**
+       * Mousetrap open/close the sidebar
+       */
+      Mousetrap.bind(['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'], function() {
+        $scope.openSearchSidebar();
+      });
+      Mousetrap.bind('esc', function() {
+        $scope.closeSearchSidebar();
+      });
+
+      /**
+       * Clicking outside the search sidebar closes it
+       */
+      $(document).on("click", function() {
+        $scope.closeSearchSidebar();
+      });
+      $(document).on("click", "search-sidebar", function(ev) {
+        ev.stopPropagation();
+      });
+
+      /**
+       * Pressing ESC while in the search sidebar input closes it
+       */
+      $(document).on('keyup', 'search-sidebar input', function(ev){
+        if (ev.keyCode == 27) $scope.closeSearchSidebar();
+      });
+
+      /**
+       * Clicking a link in the sidebar closes it
+       */
+      $(document).on('click', 'search-sidebar li a', function(){
+        $scope.closeSearchSidebar();
+      });
+
+      /**
+       * Open the search sidebar
+       */
+      $scope.openSearchSidebar = function() {
+        $('body').addClass('search-sidebar-open');
+        $('search-sidebar input').focus();
+      };
+
+      /**
+       * Close the search sidebar
+       */
+      $scope.closeSearchSidebar = function() {
+        $('body').removeClass('search-sidebar-open');
+        $('search-sidebar input').val('');
+      };
+
+      /**
+       * When the search term changes
+       */
+      $scope.termChanged = function() {
+        if ($scope.term == "") {
+          $scope.count = 0;
+          $scope.morePages = false;
+          $scope.searching = false;
+          $scope.page = 1;
+
+        } else {
+          $scope.performSearch();
+        }
+      };
+
+      /**
+       * Perform the search and render the results etc.
+       */
+      $scope.performSearch = function() {
+        $scope.searching = true;
+
+        Search.query({ term: $scope.term }).then(function(resp){
+          $scope.count = resp.meta.total;
+          $scope.morePages = resp.meta.has_more;
+          $scope.page = 1;
+
+          $scope.renderResults(resp.search_documents);
```

Comment:
Would do this the angular way i.e. having `$scope.searchResults = resp.search_documents` and then in the template for this directive looping over those. I think to do the polymorphic templates you can use `ng-switch` and you could make a directive for each template type and then using ng-switch on the class name to load a different directive/template for each type of search result.